src/core/processing/qgsprocessingparameters.cpp

```diff
@@ -2362,11 +2377,29 @@ QString QgsProcessingParameterFile::asPythonString( const QgsProcessing::PythonO
   return QString();
 }
 
+void QgsProcessingParameterFile::setExtension( const QString &extension )
+{
+  mExtension = extension;
+  mFileFilter.clear();
+}
+
+QString QgsProcessingParameterFile::fileFilter() const
+{
+  return mFileFilter;
+}
+
+void QgsProcessingParameterFile::setFileFilter( const QString &filter )
+{
+  mFileFilter = filter;
```

Comment:
Should the empty string (i.e. `fileFilter.isEmpty() ? QObject::tr( "All files (*.*)" `) : fileFilter) logic you use in the constructor be applied here too?

Reply:
I think if someone explicitly calls this with an empty string, they must have a good reason for it :wink: